Repository/DynamicRepository.php

```diff
@@ -3,10 +3,91 @@
 namespace L91\Sulu\Bundle\FormBundle\Repository;
 
 use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\NoResultException;
+use Doctrine\ORM\QueryBuilder;
 
 /**
  * Repository for the dynamic entity.
  */
 class DynamicRepository extends EntityRepository
 {
+    /**
+     * @param array $filters
+     * @param array $sort
+     * @param int|null $limit
+     * @param int|null $offset
+     *
+     * @return array
+     */
+    public function getByFilters($filters, $sort = [], $limit = null, $offset = null)
+    {
+        $queryBuilder = $this->createQueryBuilder('dynamic');
+        $fromDate = null;
+        $toDate = null;
+
+        if (isset($filters['fromDate'])) {
+            $fromDate = $filters['fromDate'];
+            unset($filters['fromDate']);
+        }
+
+        if (isset($filters['toDate'])) {
+            $toDate = $filters['toDate'];
+            unset($filters['toDate']);
+        }
+
+        // Add filters.
+        $this->addDateRangeFilter($queryBuilder, $fromDate, $toDate);
+
+        $counter = 0;
+        foreach ($filters as $key => $value) {
+            $queryBuilder->andWhere($queryBuilder->expr()->eq('dynamic.' . $key, ':value' . $counter));
+            $queryBuilder->setParameter('value' . $counter, $value);
+
+            ++$counter;
+        }
+
+        // Add sorting.
+        if (!empty($sort)) {
+            foreach ($sort as $key => $value) {
+                $queryBuilder->addOrderBy('dynamic.' . $key, $value);
+            }
+        }
+
+        // Add limit and offset.
+        if ($limit) {
+            $queryBuilder->setMaxResults($limit);
+        }
+
+        if ($offset) {
+            $queryBuilder->setFirstResult($offset);
+        }
+
+        try {
+            return $queryBuilder->getQuery()->getResult();
+        } catch (NoResultException $exception) {
```

Comment:
this try and catch is not needed as it is not a `getSingleResult` so it will not throw a NoResultException